atom/src/validatebehavior.cpp

```diff
@@ -682,6 +682,25 @@ float_range_handler( Member* member, CAtom* atom, PyObject* oldvalue, PyObject*
 }
 
 
+PyObject*
+float_range_promote_handler( Member* member, CAtom* atom, PyObject* oldvalue, PyObject* newvalue )
+{
+    if( PyFloat_Check( newvalue ) )
+        return float_range_handler(member, atom, oldvalue, newvalue);
+    if( PyLong_Check( newvalue ) )
+    {
+        double value = PyLong_AsDouble( newvalue );
+        if( value == -1.0 && PyErr_Occurred() )
+            return 0;
+        PyObject* convertedvalue = PyFloat_FromDouble( value );
+        PyObject* result = float_range_handler(member, atom, oldvalue, convertedvalue);
+        cppy::decref(convertedvalue);
+        return result;
```

Comment:
You can use cppy smart pointer to simplify your life here.

```suggestion
        cppy::ptr convertedvalue( PyFloat_FromDouble( value ) );
        return float_range_handler( member, atom, oldvalue, convertedvalue.get() );
```